### PR TITLE
AB#5012

### PR DESCRIPTION
**Pull request by sachin755**
### Proposed changes

*
*

### Related issues

*
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by johnjdl** [comment]
This is working locally, @DallasP9124 and @r-joyce  please review and merge. @r-joyce @CyberDaedalus00 can we bump develop as soon as this is merged. I will be testing on develop before closing out the tickets going forward.

File: opencti-platform/opencti-front/src/private/components/dataEntities/DataSources.js [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+  handleClearSelectedElements() {
+    this.setState({ selectAll: false, selectedElements: null });
+  }
+
+  handleRiskCreation() {
+    this.setState({ openDataCreation: true });
+  }
+
+  handleRefresh() {
+    this.props.history.push('/data/data source');
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/DataSources.js:92]: `handleRefresh` doesnt appear to be used, if that is the case can you please remove this?

File: opencti-platform/opencti-front/src/private/components/dataEntities/Entities.js [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    this.setState({ selectAll: false, selectedElements: null });
+  }
+
+  handleRiskCreation() {
+    this.setState({ openDataCreation: true });
+  }
+
+  handleRefresh() {
+    this.props.history.push('/data/entities');
+  }
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/Entities.js:93]: @sachin755 looks to not be used. If that is intentional, would you please remove it?

File: opencti-platform/opencti-front/src/private/components/dataEntities/Root.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          component={DataSources}
+        />
+        {/* <BoundaryRoute
+          path="/data/entities/:entityId"
+          render={(routeProps) => <RootDevice {...routeProps} me={me} />}
+        />
+        <BoundaryRoute
+          path="/data/data source/:dataSourceId"
+          render={(routeProps) => <RootDevice {...routeProps} me={me} />}
+        /> */}
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/Root.js:35]: @sachin755 if their is no intended future use of these routes would you please remove these?

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+        input: finalValues,
+      },
+      setSubmitting,
+      onCompleted: (data) => {
+        setSubmitting(false);
+        resetForm();
+        this.handleClose();
+        // this.props.history.push('/activities/risk assessment/risks');
+      },
+      onError: (err) => console.error(err),
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js:177]: @sachin755 please apply an error toast here.

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+  // last_seen: Yup.date()
+  //   .nullable()
+  //   .typeError(t('The value must be a date (YYYY-MM-DD)')),
+  // sophistication: Yup.string().nullable(),
+  // resource_level: Yup.string().nullable(),
+  // primary_motivation: Yup.string().nullable(),
+  // secondary_motivations: Yup.array().nullable(),
+  // personal_motivations: Yup.array().nullable(),
+  // goals: Yup.string().nullable(),
+});
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js:125]: @sachin755 regarding the large chunks of commented out code, if there is no intent for them to be used as of now, please remove them to cleanup the code. I will use this as an overall request for all commented code on this PR.

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceDeletion.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+      mutation: DataSourceDeletionDarkLightMutation,
+      variables: {
+        id: this.props.id,
+      },
+      onCompleted: (data) => {
+        this.setState({ deleting: false });
+        this.handleClose();
+        // this.props.history.push('/activities/risk assessment/risks');
+      },
+      onError: (err) => console.error(err),
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceDeletion.js:152]: @sachin755 please apply an error toast here.

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/EntitiesCreation.js [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+      created: values.riskDetailsCreated,
+      // remediations: {
+      //   response_type: values.response_type,
+      //   lifecycle: values.lifecycle,
+      // }
+    };
+    const finalValues = R.pipe(
+      R.assoc('name', values.name),
+    )(adaptedValues);
+    console.log('RiskCreationFinal', finalValues);
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/data/EntitiesCreation.js:169]: @sachin755 please remove the console.log

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/EntitiesCreation.js [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+        input: finalValues,
+      },
+      setSubmitting,
+      onCompleted: (data) => {
+        setSubmitting(false);
+        resetForm();
+        this.handleClose();
+        // this.props.history.push('/activities/risk assessment/risks');
+      },
+      onError: (err) => console.error(err),
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/data/EntitiesCreation.js:182]: @sachin755 please apply an error toast.

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/EntitiesDeletion.js [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+      mutation: EntitiesDeletionDarkLightMutation,
+      variables: {
+        id: this.props.id,
+      },
+      onCompleted: (data) => {
+        this.setState({ deleting: false });
+        this.handleClose();
+        // this.props.history.push('/activities/risk assessment/risks');
+      },
+      onError: (err) => console.error(err),
```
Comment (BigBallard) [inline opencti-platform/opencti-front/src/private/components/dataEntities/data/EntitiesDeletion.js:152]: @sachin755 please add an error toast.

**Review by BigBallard — CHANGES_REQUESTED** [review]
@sachin755 most of the requested changes refer to console log, addition of error toasts, some questions for clarification, and about commented out code. Generally we try to reduce the amount of commented out code for only code that is highly anticipated to be used in the future. If it is not needed please make sure to clean them up. Thank you!
@johnjdl fyi

File: opencti-platform/opencti-front/src/private/components/dataEntities/DataSources.js (hunk 1 above)
Comment (AMAN-SHARMA-82591) [reply opencti-platform/opencti-front/src/private/components/dataEntities/DataSources.js:92]: @DallasP9124  Most of the code that is being commented is going to be used in future. Please follow AB#5013 PR, in which we've merged AB#5012 code and used all of the commented code. Most of the console.error/bakedToast changes are going to be visible in #302 .

File: opencti-platform/opencti-front/src/private/components/dataEntities/Root.js (hunk 3 above)
Comment (AMAN-SHARMA-82591) [reply opencti-platform/opencti-front/src/private/components/dataEntities/Root.js:35]: @DallasP9124  It's for the Data source page that is going to get implemented in future.

File: opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js (hunk 4 above)
Comment (AMAN-SHARMA-82591) [reply opencti-platform/opencti-front/src/private/components/dataEntities/data/DataSourceCreation.js:177]: #299 is just for the design of main screen.  The chages you've suggested are going to be inside #302

**Comment by johnjdl** [comment]
@DallasP9124 the changes requested are in 5013, @r-joyce can we go ahead and merge this PR *5012 before 5013 which I am requesting today as well.

**Review by BigBallard — APPROVED** [review]

**Comment by BigBallard** [comment]
@johnjdl @r-joyce good assuming fixes in [AB#5013](https://dev.azure.com/DkLt/1f741bf4-fc05-4308-8389-99678c3c5ab2/_workitems/edit/5013)

**Review by r-joyce — APPROVED** [review]

**Pull request merged**